lib/rails_cursor_pagination/cursor.rb

```diff
@@ -42,18 +42,31 @@ def decode(encoded_string:, order_field: :id)
           end
           new(id: decoded, order_field: :id)
         else
-          unless decoded.is_a?(Array) && decoded.size == 2
-            raise InvalidCursorError,
-                  "The given cursor `#{encoded_string}` was decoded as " \
-                  "`#{decoded}` but could not be parsed"
-          end
-          new(id: decoded[1], order_field: order_field,
-              order_field_value: decoded[0])
+          decode_custom_order_field(decoded: decoded, order_field: order_field)
         end
       rescue ArgumentError, JSON::ParserError
         raise InvalidCursorError,
               "The given cursor `#{encoded_string}` could not be decoded"
       end
+
+      def decode_custom_order_field(decoded:, order_field:)
+        unless decoded.is_a?(Array) && decoded.size == 2
+          raise InvalidCursorError,
+                "The given cursor `#{encoded_string}` was decoded as " \
+                "`#{decoded}` but could not be parsed"
+        end
+        if decoded[0].is_a?(Hash)
```

Comment:
`decoded[0]` could be turned into a hash for reasons other than this one right? So is it worth adding a check here for `seconds` and `nanoseconds` keys?

```ruby
if decoded[0].is_a?(Hash) && ['seconds', 'nanoseconds'].all? { |key| decoded[0].key? key }
```